src/uct/tcp/tcp_net.c

```diff
@@ -26,12 +26,48 @@
 
 typedef ssize_t (*uct_tcp_io_func_t)(int fd, void *data, size_t size, int flags);
 
+/* Caller is responsible to free memory allocated if str_addr wasn't provided */
+char *uct_tcp_sockaddr_2_string(const struct sockaddr_in *addr, char **str_addr,
+                                size_t *str_addr_len)
+{
+    char *tmp_addr = NULL;
+    int ret;
+
+    if (str_addr != NULL && *str_addr != NULL && str_addr_len && NULL) {
```

Comment:
wrap every condition into ()

Reply:
if ((str_addr != NULL) && (*str_addr != NULL) && str_addr_len && NULL

BTW, && NULL - what is this?

Reply:
> BTW, && NULL - what is this?

This should be `str_addr_len == NULL`

> wrap every condition into ()

fixed

Reply:
hmmm, you changed it to
```
str_addr_len != NULL
```
is it ok?

Reply:
Sorry, the previous comment wasn't correct:

> This should be `str_addr_len == NULL`

I meant `str_addr_len != NULL`

So yes, the change corresponds to what I had in mind :)